server/services/Users.js

```diff
@@ -68,6 +74,102 @@ class Users {
     if (!usersReturning) throw new Error(THE_USER_IS_NOT_FOUND);
     return this._db.events.findByPk(usersReturning.uid, {attributes: ['one_page_token']});
   }
+
+  /**
+   * @description Attempt to send an email.
+   * Follow the SPEC, an email is possible to map to many users.
+   * Send emails to all users in this email.
+   * Every email with uid has a cool-down period of 300 seconds. Format: EMAIL:COOL_DOWN:<EMAIL>
+   * @param {String} email The user email
+   */
+  async sendEmail(email) {
+    const usersArray = await this._db.users.findAll({
+      where: {email},
+      attributes: ['uid', 'nick_name'],
+      include: {
+        model: this._db.events,
+        attributes: ['one_page_token'],
+        required: false
+      }});
+    if (!usersArray.length) throw new Error(THE_USER_IS_NOT_FOUND);
+
+    // Check email cool down status. If the email is cool down, then throw error.
+    const emailCoolDownCheck = await this.redisClient.get(`EMAIL:COOL_DOWN:${email}`);
```

Comment:
Consider combining the get and set here with SET NX or similar atomic command so that there's no gap / race condition for the attacker to trigger multiple email send before `redisClient.set` is called.